### PR TITLE
do not patch status if resource is not managed

### PR DESCRIPTION
**Pull request by surajkota**
Issue #, if available:
https://github.com/aws-controllers-k8s/community/issues/931

Description of changes:
Solution 1 in the issue linked above - `return nil, nil` instead of `resource, nil` after finalizer is removed 

Testing:
`make test`
Manually using sagemaker controller

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/runtime/reconciler.go
Comment (vijtrip2) [inline pkg/runtime/reconciler.go:514]: Please update this comment to say exactly what is happening here.
"Return 'nil' resource so that patchStatus attempt is not made. Underlying K8s resource gets deleted after removing finalizer string and patchStatus call will return NotFound error."

**Review by vijtrip2 — APPROVED** [review]

File: pkg/runtime/reconciler.go
Comment (surajkota) [reply pkg/runtime/reconciler.go]: done!

**Review by vijtrip2 — APPROVED** [review]

**Comment by vijtrip2** [comment]
Waiting for other ACK core team members to take a look as well. 
I will merge this PR tomorrow morning, if it does not receive new comments.

File: pkg/runtime/reconciler.go
Comment (jaypipes) [inline pkg/runtime/reconciler.go:520]: Instead of all of the above changes, would it not be easier to just change [this one line](https://github.com/aws-controllers-k8s/runtime/blob/24e0b8ddeaaf7df4bf55ef1a75e4608ea5ec01fb/pkg/runtime/reconciler.go#L647)

from this:

```go
	if ackcompare.IsNotNil(latest) {
```

to this:

```go
	if ackcompare.IsNotNil(latest) && r.rd.IsManaged(latest) {
```

Comment (surajkota) [reply pkg/runtime/reconciler.go:520]: that was my 3rd solution on the issue. What do you think about doing this in addition to the current change?
so logically the return from deleteResource also matches the condition change you suggested

Comment (jaypipes) [reply pkg/runtime/reconciler.go:520]: @surajkota the return from the Delete operation should be the latest observed state of the resource. Not nil.

Comment (surajkota) [reply pkg/runtime/reconciler.go]: oh right, done!

**Review by jaypipes — APPROVED** [review]
:+1:

**Pull request merged**
